behaviour/graql/language/delete.feature

```diff
@@ -16,12 +33,566 @@
+      """
+      match (friend: $x, friend: $y) isa friendship; get;
+      """
+    Then uniquely identify answer concepts
+      | x    | y    |
+      | P2   | P3   |
+      | P3   | P2   |
+
+
+  Scenario: delete a role player from a relation using meta role removes the player from the relation
+    When graql insert
+      """
+      insert
+      $x isa person, has ref 0;
+      $y isa person, has ref 1;
+      $z isa person, has ref 2;
+      $r (friend: $x, friend: $y, friend: $z) isa friendship,
+         has ref 3;
+      """
+    When the integrity is validated
+
+    Then concept identifiers are
+      |      | check | value     |
+      | P1   | key   | ref:0     |
+      | P2   | key   | ref:1     |
+      | P3   | key   | ref:2     |
+      | FR   | key   | ref:3     |
+
+    Then graql delete
+      """
+      match
+        $r (friend: $x, friend: $y, friend: $z) isa friendship;
+        $x isa person, has ref 0;
+        $y isa person, has ref 1;
+        $z isa person, has ref 2;
+      delete
+        $r (role: $x);
+      """
+
+    Then get answers of graql query
+      """
+      match (friend: $x, friend: $y) isa friendship; get;
+      """
+    Then uniquely identify answer concepts
+      | x    | y    |
+      | P2   | P3   |
+      | P3   | P2   |
+
+
+  Scenario: delete an instance removes it from all relations
+    When graql insert
+      """
+      insert
+      $x isa person, has ref 0;
+      $y isa person, has ref 1;
+      $z isa person, has ref 2;
+      $r (friend: $x, friend: $y) isa friendship, has ref 3;
+      $r2 (friend: $x, friend: $z) isa friendship, has ref 4;
+      """
+    When the integrity is validated
+
+    Then concept identifiers are
+      |      | check | value     |
+      | P1   | key   | ref:0     |
+      | P2   | key   | ref:1     |
+      | P3   | key   | ref:2     |
+      | FR1  | key   | ref:3     |
+      | FR2  | key   | ref:4     |
+
+    Then graql delete
+      """
+      match
+        $x isa person, has ref 0;
+      delete
+        $x isa person;
+      """
+
+    Then get answers of graql query
+      """
+      match $r (friend: $x) isa friendship; get;
+      """
+    Then uniquely identify answer concepts
+      | r     | x    |
+      | FR1   | P2   |
+      | FR2   | P3   |
+
+
+  Scenario: delete duplicate role players from a relation removes duplicate player from relation
+    When graql insert
+      """
+      insert
+      $x isa person, has ref 0;
+      $y isa person, has ref 1;
+      $r (friend: $x, friend: $x, friend: $y) isa friendship, has ref 2;
+      """
+    When the integrity is validated
+
+    Then concept identifiers are
+      |      | check | value     |
+      | P1   | key   | ref:0     |
+      | P2   | key   | ref:1     |
+      | FR   | key   | ref:2     |
+
+    Then graql delete
+      """
+      match
+        $r (friend: $x, friend: $x) isa friendship;
+      delete
+        $r (friend: $x, friend: $x);
+      """
```

Comment:
Do we have the case for 
```
match
$r (friend: $x, friend: $x) isa friendship;
delete
$r (friend: $x);
```
?

Reply:
Also this one:
```
match
$r (friend: $x) isa friendship;
delete
$r (friend: $x);
```
But where `$x` plays friend twice as in this example

Reply:
good ideas!